react-basic-practice/src/pages/HomePage/index.jsx

```diff
@@ -14,26 +21,76 @@ const HomePage = () => {
   const [order, setOrder] = useState('');
   const [fieldSort, setFieldSort] = useState('');
   const [valueSelected, setValueSelected] = useState('');
+  const [products, setProducts] = useState([]);
+  const [pageNumber, setPageNumber] = useState(1);
+  const [isDisabled, setIsDisabled] = useState(false);
+  const { loading, setLoading } = useLoading();
+
+  useEffect(() => {
+    const getData = async () => {
+      try {
+        setLoading(true);
+        const data = await getListProducts(pageNumber, keyword, fieldSort, order);
+        data ? setProducts(data) : setProducts([]);
+        data.length >= 6 ? setIsDisabled(false) : setIsDisabled(true);
+      } catch {
+        alert('Error loading data, please reload the page');
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    getData();
+  }, [keyword, fieldSort, order]);
+
+  useEffect(() => {
+    const getData = async () => {
+      try {
+        setLoading(true);
+        const data = await getListProducts(pageNumber, keyword, fieldSort, order);
+        const newData = [...products, data];
+
+        data ? setProducts(newData.flat()) : setProducts([]);
```

Comment:
A lot of duplicate code with above useEffect. Please think of a way to write it once.